### PR TITLE
Add snapshot tests for IdCompressor

### PR DESCRIPTION
**Pull request by justus-camp**
## Description
At the moment if you made a breaking change to IdCompressor's summary there aren't any tests to detect that happening. This adds a test suite that would fail if the summary format changed. The helpers to write to files are copy-pasted from tree2's similar suite rather than pulling them out to a shared package.

## Reviewer Guidance
The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: packages/runtime/id-compressor/src/test/snapshots/summary.spec.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+ */
+
+/* eslint-disable import/no-nodejs-modules */
+
+import { strict as assert } from "assert";
+import { existsSync, rmSync, mkdirSync, writeFileSync, readFileSync } from "fs";
+import path from "path";
+import { SessionId, createIdCompressor } from "../..";
+
+const regenerateSnapshots = process.argv.includes("--snapshot");
```
Comment (taylorsw04) [inline packages/runtime/id-compressor/src/test/snapshots/summary.spec.ts:13]: should there be something in the package.json for this?

Comment (taylorsw04) [reply packages/runtime/id-compressor/src/test/snapshots/summary.spec.ts:13]: also, probably fine but just checking argv without knowing the command is a bit scary

Comment (justus-camp) [reply packages/runtime/id-compressor/src/test/snapshots/summary.spec.ts:13]: Added a script to the package.json for this. As for blindly checking argv, I just followed what we do in tree's directory but am open if you want to do it a diff way.

**Comment by justus-camp** [comment]
> Did you dry-run making a persisted change? If so, I'm happy with this.

Yeah I changed the ordering of some of the serialization and it correctly fails the tests.

**Review by taylorsw04 — APPROVED** [review]

**Pull request merged**